skyllh/core/scrambling.py

```diff
@@ -234,3 +234,108 @@ def scramble_data(self, rss, data, copy=False):
         data = self._method.scramble(rss, data)
 
         return data
+
+
+class TimeDepScrambling(DataScramblingMethod):
+    """The TimeScramblingMethod class provides a data scrambling method to
+        perform data coordinate scrambling based on a generated time. It draws a
+        random time from a time generator and transforms the horizontal (local)
+        coordinates into equatorial coordinates using a specified transformation
+        function.
+        """
+    def __init__(self, events):
+            """Initializes a new time scrambling instance.
+
+            Parameters
+            ----------
+            timegen : TimeGenerator
+                The time generator that should be used to generate random MJD times.
+            hor_to_equ_transform : callable
+                The transformation function to transform coordinates from the
+                horizontal system into the equatorial system.
+
+                The call signature must be:
+
+                    __call__(azi, zen, mjd)
+
+                The return signature must be: (ra, dec)
+
+            """
+            super(TimeDepScrambling, self).__init__()
+
+            # get the times and the events for this for correct weighting
+            self.weights = []
+            for start, stop in zip(events.grl['start'], events.grl['stop']):
+                mask_grl = (events.exp["time"] > start) & (events.exp['time'] < stop)
+                # how many events in each run (relative to all events)
+                self.weights.append(len(events.exp[mask_grl]) / len(events.exp['time']))
+            
+            # renormalize
+            weight_sum = sum(self.weights)
+            self.weights = [x / weight_sum for x in self.weights]
```

Comment:
What is the reasoning for these weights?

Reply:
To preserve the number of events per run. This preserves e.g. seasonal variations.